e2e/single-cluster/oci_registry_test.go

```diff
@@ -0,0 +1,427 @@
+		}
+
+		// purge content id if needed
+		if contentIDToPurge != "" {
+			out, err := k.Delete("contents", contentIDToPurge)
+			Expect(out).To(ContainSubstring("deleted"))
+			Expect(err).ToNot(HaveOccurred())
+		}
+	})
+
+	When("creating a gitrepo resource with ociRegistry info", func() {
+		Context("containing a valid helm chart", func() {
+			BeforeEach(func() {
+				asset = "single-cluster/test-oci.yaml"
+				insecureSkipTLS = true
+				forceGitRepoUrl = ""
+				experimentalValue = true
+			})
+
+			It("creates the bundle", func() {
+				Eventually(func() string {
+					out, _ := k.Namespace("fleet-local").Get("bundles")
+					return out
+				}).Should(ContainSubstring("sample-simple-chart-oci"))
+			})
+			It("sets the ContentsID field in the bundle", func() {
+				Eventually(func() string {
+					contentsID, _ = k.Namespace("fleet-local").Get("bundle", "sample-simple-chart-oci", `-o=jsonpath={.spec.contentsId}`)
+					return contentsID
+				}).Should(ContainSubstring("s-"))
+			})
+			It("sets Resource key with the oci path to the manifest", func() {
+				Eventually(func() bool {
+					out, _ := k.Namespace("fleet-local").Get("bundle", "sample-simple-chart-oci", `-o=jsonpath={.status.resourceKey}`)
+					var existingResourceKeys []fleetv1.ResourceKey
+					err := json.Unmarshal([]byte(out), &existingResourceKeys)
+					if err != nil {
+						return false
+					}
+					// checks that it only creates 1 resource key
+					if len(existingResourceKeys) != 1 {
+						return false
+					}
+					if existingResourceKeys[0].Name != fmt.Sprintf("oci://%s/%s:latest", ociRegistry, contentsID) {
+						return false
+					}
+					return true
+				}).Should(BeTrue())
+			})
+			It("creates a bundle secret", func() {
+				Eventually(func() string {
+					out, err := k.Namespace("fleet-local").Get("secret", contentsID)
+					if err != nil {
+						// return nothing in case of error
+						// This avoids false positives when kubectl returns "secrets "XXXXX" not found
+						return ""
+					}
+					return out
+				}).Should(ContainSubstring(contentsID))
+			})
+			It("creates a bundledeployment secret", func() {
+				Eventually(func() string {
+					out, err := k.Namespace(downstreamNamespace).Get("secret", contentsID)
+					if err != nil {
+						// return nothing in case of error
+						// This avoids false positives when kubectl returns "secrets "XXXXX" not found
+						return ""
+					}
+					return out
+				}).Should(ContainSubstring(contentsID))
+			})
+			It("does not deploy a contents resource for this chart", func() {
+				out, _ := k.Get("contents")
+				Expect(out).NotTo(ContainSubstring(contentsID))
+			})
+			It("deploys the helm chart", func() {
+				Eventually(func() string {
+					out, _ := k.Namespace("fleet-local").Get("configmaps")
+					return out
+				}).Should(ContainSubstring("sample-config"))
+			})
+		})
+	})
+
+	When("creating a gitrepo resource with invalid ociRegistry info and experimental flag is false", func() {
```

Comment:
2 questions here:
* why is it relevant for OCI registry info to be invalid here, if the experimental flag is `false` anyway? Wouldn't the test make more sense with valid registry info?
* do we actually provide invalid registry info to this test case? :eyes:

Reply:
1. That test is checking that we're not doing any check about wrong OCI registry values when the flag is not set.
2. You're absolutely right. It should still pass now that I changed it to be using a non-valid url

Reply:
I see, so this wasn't a typo in the test description. This works for me, thanks for the update.